plugins/modules/ecs_taskdefinition.py

```diff
@@ -796,28 +817,31 @@ def main():
                 module.fail_json(msg='links parameter is not supported if network mode is awsvpc.')
 
             for environment in container.get('environment', []):
-                environment['value'] = to_text(environment['value'])
+                environment['value'] = environment['value']
 
             for environment_file in container.get('environmentFiles', []):
                 if environment_file['value'] != 's3':
                     module.fail_json(msg='The only supported value for environmentFiles is s3.')
 
             for linux_param in container.get('linuxParameters', {}):
-                if linux_param.get('devices') and launch_type == 'FARGATE':
+                if linux_param == 'maxSwap' and launch_type == 'FARGATE':
                     module.fail_json(msg='devices parameter is not supported with the FARGATE launch type.')
 
-                if linux_param.get('maxSwap') and launch_type == 'FARGATE':
+                if linux_param == 'maxSwap' and launch_type == 'FARGATE':
                     module.fail_json(msg='maxSwap parameter is not supported with the FARGATE launch type.')
-                elif linux_param.get('maxSwap') and linux_param['maxSwap'] < 0:
+                elif linux_param == 'maxSwap' and int(container['linuxParameters']['maxSwap']) < 0:
```

Comment:
What happen if key `maxSwap` does not exist? Is it possible? I've no idea.

Reply:
same for `swappiness`

Reply:
@markuman `maxSwap` and `swappiness` are optional and they can be specified within `linuxParameters`. If this is what you meant. This flow here is only executed when the parameters are specified.